pimd/pim_cmd_common.c

```diff
@@ -653,3 +673,1302 @@ int pim_process_no_rp_plist_cmd(struct vty *vty, const char *rp_str,
+
+			if (json_object_object_get_ex(
+				    val, "pimDesignatedRouterLocal",
+				    &json_tmp)) {
+				vty_out(vty, "%15s  ", "local");
+			} else {
+				json_object_object_get_ex(
+					val, "pimDesignatedRouter", &json_tmp);
+				vty_out(vty, "%15s  ",
+					json_object_get_string(json_tmp));
+			}
+
+			json_object_object_get_ex(val, "firstHopRouter",
+						  &json_tmp);
+			vty_out(vty, "%3d  ", json_object_get_int(json_tmp));
+
+			json_object_object_get_ex(val, "pimIfChannels",
+						  &json_tmp);
+			vty_out(vty, "%9d\n", json_object_get_int(json_tmp));
+		}
+	}
+}
+
+void pim_show_interfaces_single(struct pim_instance *pim,
+				struct vty *vty, const char *ifname,
+				bool mlag, json_object *json)
+{
+	pim_addr ifaddr;
+	struct interface *ifp;
+	struct listnode *neighnode;
+	struct pim_interface *pim_ifp;
+	struct pim_neighbor *neigh;
+	struct pim_upstream *up;
+	time_t now;
+	char dr_str[PIM_ADDRSTRLEN];
+	char dr_uptime[10];
+	char expire[10];
+	char grp_str[PIM_ADDRSTRLEN];
+	char hello_period[10];
+	char hello_timer[10];
+	char neigh_src_str[PIM_ADDRSTRLEN];
+	char src_str[PIM_ADDRSTRLEN];
+	char stat_uptime[10];
+	char uptime[10];
+	int found_ifname = 0;
+	int print_header;
+	json_object *json_row = NULL;
+	json_object *json_pim_neighbor = NULL;
+	json_object *json_pim_neighbors = NULL;
+	json_object *json_group = NULL;
+	json_object *json_group_source = NULL;
+	json_object *json_fhr_sources = NULL;
+	struct pim_secondary_addr *sec_addr;
+	struct listnode *sec_node;
+
+	now = pim_time_monotonic_sec();
+
+	FOR_ALL_INTERFACES (pim->vrf, ifp) {
+		pim_ifp = ifp->info;
+
+		if (!pim_ifp)
+			continue;
+
+		if (mlag == true && pim_ifp->activeactive == false)
+			continue;
+
+		if (strcmp(ifname, "detail") && strcmp(ifname, ifp->name))
+			continue;
+
+		found_ifname = 1;
+		ifaddr = pim_ifp->primary_address;
+		snprintfrr(dr_str, sizeof(dr_str), "%pPAs",
+			   &pim_ifp->pim_dr_addr);
+		pim_time_uptime_begin(dr_uptime, sizeof(dr_uptime), now,
+				      pim_ifp->pim_dr_election_last);
+		pim_time_timer_to_hhmmss(hello_timer, sizeof(hello_timer),
+					 pim_ifp->t_pim_hello_timer);
+		pim_time_mmss(hello_period, sizeof(hello_period),
+			      pim_ifp->pim_hello_period);
+		pim_time_uptime(stat_uptime, sizeof(stat_uptime),
+				now - pim_ifp->pim_ifstat_start);
+
+		if (json) {
+			char pbuf[PREFIX2STR_BUFFER];
+
+			json_row = json_object_new_object();
+			json_object_pim_ifp_add(json_row, ifp);
+
+			if (!pim_addr_is_any(pim_ifp->update_source)) {
+				json_object_string_addf(
+					json_row, "useSource", "%pPAs",
+					&pim_ifp->update_source);
+			}
+			if (pim_ifp->sec_addr_list) {
+				json_object *sec_list = NULL;
+
+				sec_list = json_object_new_array();
+				for (ALL_LIST_ELEMENTS_RO(
+					     pim_ifp->sec_addr_list, sec_node,
+					     sec_addr)) {
+					snprintfrr(pbuf, sizeof(pbuf), "%pFXs",
+						   &sec_addr->addr);
+					json_object_array_add(
+						sec_list,
+						json_object_new_string(pbuf));
```

Comment:
`json_object_new_stringf` to remove extra formatting step

Reply:
Resolved